### PR TITLE
fix(lib): validate every variadic fallback

### PR DESCRIPTION
**Pull request by jdx**
## Summary

- normalize variadic environment, default, and conditional-default values to `MultiString`
- split fallback values with the declared delimiter before choices and expression validation
- validate later fallback values instead of only the first or a combined string

## Context

Follow-up to #1037 for the final incremental CodeRabbit finding that arrived as the original PR merged.

## Test plan

- `cargo test --workspace --all-features`
- `cargo clippy --workspace --all-features -- -D warnings`
- regression coverage includes positional env values plus flag env, default, and `default_if` values

_This PR description was AI-generated._

---

> [!NOTE]
> **Medium Risk**
> Changes core CLI fallback parsing and validation semantics; incorrect edge cases could affect apps using delimited env/defaults, though behavior is intentionally stricter with added tests.
> 
> **Overview**
> **Fixes variadic fallbacks** (env, defaults, `default_if`) so delimited strings are split and **each** piece is validated—not only the first value or one combined string.
> 
> Fallback binding now uses `split_fallback_values` with the arg’s delimiter, runs `validate_choice_values` on the full list, enforces `var_min`/`var_max` via new `validate_*_fallback_count` helpers, and stores `MultiString` when `var` applies (including flag env paths where `flag.var` or `arg.var` is set). `bind_flag_fallback` is aligned so option flags with variadic args split and validate the same way; boolean-only variadic flags still map to `MultiBool`.
> 
> Regression tests cover delimiter-split env/defaults, per-value expression errors, and `var_max` overflow messages for positionals and flags.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 44d5cd52d768bd4718d0e5ef61d049d64bfa5951. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

## Summary by CodeRabbit

* **Bug Fixes**
  * Environment-variable and default fallback values are now correctly split using configured delimiters before validation.
  * Variadic arguments and flags now preserve their expected multi-value output when fallbacks contain delimited values.
  * Every split fallback value is validated, providing clearer error reporting for invalid entries.

### CODE REVIEW

File: lib/src/parse.rs [hunk 1]
```diff
@@ -471,14 +476,20 @@ impl<'a> Parser<'a> {
...
                             custom_env,
                         )?;
-                        out.flags
-                            .insert(Arc::clone(flag), ParseValue::String(env_value));
+                        let parsed = if flag.var || arg.var {
+                            ParseValue::MultiString(values)
+                        } else {
+                            ParseValue::String(values.into_iter().next().unwrap_or_default())
+                        };
+                        out.flags.insert(Arc::clone(flag), parsed);
```
[the remark this answers is not in the document]
Comment (jdx) [reply lib/src/parse.rs:505]: Fixed in 44d5cd52. Split env, default, and default_if values now enforce positional, repeated-flag, and nested variadic argument bounds after fallback binding, with regression coverage for each path. _This reply was AI-generated._
